### DISCUSSION TITLE
Copilot's IntelliJ plugin can't connect to GitHub's servers

### DISCUSSION THREAD

**Discussion by MAXOUXAX** [community/community#17276] [Copilot Conversations]
EDIT: Using Copilot 1.1.23.1570

I've been using Copilot just fine for a couple of weeks on my school computer's, but 3 days ago, when I logged back in my IDE (IntelliJ), Copilot wasn't working at all, even though it wasn't showing disconnected.

I therefore tried to uninstall the plugin from IntelliJ, restart IntelliJ, and install the plugin again. Tried logging in, and after entering the device code on the auth website, went back to IntelliJ, only to find the dialog loading for a few seconds, before closing and showing an error saying that Copilot couldn't authentificate.

I went to the logs to try and see what went wrong, and found a network error from Copilot, the error message stating the connection timed out. When opening the https://IP:443 URL in my browser, the connection worked correctly (of course showing an invalid request error from GitHub servers).

After seeing the timed out error, I tried uninstalling Copilot again, but also deleting the plugin config files in the `user\.IntelliJIdea\config\options` directory completely, before restarting the computer and trying again. That sadly did not fix my issue.

EDIT 2: After hitting the Login to GitHub button for the first time, the popup with the device code opens up correctly, even though as explained the auth process doesn't work. However, after trying to login once, the next presses on the login button will open up an infinite loading dialog saying "Retrieving GitHub device code", which will display an error after a minute or so.

Now, the computer is running on my school Internet, and issues like that are hard to fix, if not completly unrelated to Copilot.
But I know students that are in my class that have successfully logged in yesterday, without any hassle, with computers from school, running on my school Internet and therefore sharing the same network rules.

I'm pretty confused about the situation, and would like to know if there are any more debugging steps I could try, or any information I could gather (and where) to help me fix the issue.

**Comment by tbenbrahim** [comment]
As of the latest updates, Copilot no longer works with proxies on IntelliJ, and older versions of the plugin do not work either, It is very unfortunate since proxies are used in most educational and business environments. Based on the response from Microsoft, they do not consider something urgent to fix, they have more shiny things to play with. 
It is a tech preview after all, maybe by the time it is a paid service, it will actually be usable by more people. As of right now, for me, it is only suitable for use at home for weekend projects,

**Reply by MAXOUXAX** [reply]
> As of the latest updates, Copilot no longer works with proxies on IntelliJ, and older versions of the plugin do not work either, It is very unfortunate since proxies are used in most educational and business environments. Based on the response from Microsoft, they do not consider something urgent to fix, they have more shiny things to play with. It is a tech preview after all, maybe by the time it is a paid service, it will actually be usable by more people. As of right now, for me, it is only suitable for use at home for weekend projects,

This doesn't seem right, since I mentioned in my message that other students I know have successfully logged in and used Copilot, without any hassle, with computers from school, running on my school Internet and therefore sharing the same network rules.
